src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -196,20 +194,108 @@ export function useSwapInputsController({
         return;
       }
 
-      inputValues.modify(prev => {
-        return {
-          ...prev,
-          outputAmount: outputAmount || prev.outputAmount,
-          outputNativeValue: outputAmountNative || prev.outputNativeValue,
-          inputAmount: inputAmount || prev.inputAmount,
-          inputNativeValue: inputAmountNative || prev.inputNativeValue,
-        };
-      });
-      // TODO: Update the inputAmount and outputAmount based on the quote
+      if (inputAmount) {
+        const price = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+        inputValues.modify(prev => {
+          return {
+            ...prev,
+            inputAmount,
+            inputNativeValue: inputAmount * price,
+          };
+        });
+      }
+
+      if (outputAmount) {
+        const price = internalSelectedOutputAsset.value?.nativePrice || internalSelectedOutputAsset.value?.price?.value || 0;
+        inputValues.modify(prev => {
+          return {
+            ...prev,
+            outputAmount,
+            outputNativeValue: outputAmount * price,
+          };
+        });
+      }
     },
-    [inputValues, internalSelectedInputAsset.value?.balance.amount, isFetching, quote, sliderXPosition, updateQuoteStore]
+    [
+      inputValues,
+      internalSelectedInputAsset.value?.balance.amount,
+      internalSelectedInputAsset.value?.nativePrice,
+      internalSelectedInputAsset.value?.price?.value,
+      internalSelectedOutputAsset.value?.nativePrice,
+      internalSelectedOutputAsset.value?.price?.value,
+      isFetching,
+      quote,
+      sliderXPosition,
+      updateQuoteStore,
+    ]
   );
 
+  const fetchAndUpdatePrices = async ({
+    inputAsset,
+    outputAsset,
+  }: {
+    inputAsset: ExtendedAnimatedAssetWithColors | null;
+    outputAsset: ExtendedAnimatedAssetWithColors | null;
+  }) => {
+    const updateNativePriceForAsset = ({ price, type }: { price: number; type: string }) => {
+      'worklet';
+      if (type === 'inputAsset') {
+        internalSelectedInputAsset.modify(prev => ({ ...prev, nativePrice: price }));
+      } else if (type === 'outputAsset') {
+        internalSelectedOutputAsset.modify(prev => ({ ...prev, nativePrice: price }));
+      }
+    };
+
+    const getAssetNativePrice = async ({ asset, type }: { asset: ExtendedAnimatedAssetWithColors | null; type: string }) => {
+      if (!asset) return;
+
+      const address = asset.address;
+      const network = ethereumUtils.getNetworkFromChainId(asset.chainId);
+      const currency = store.getState().settings.nativeCurrency;
+
+      try {
+        const tokenData = await fetchExternalToken({
+          address,
+          network,
+          currency,
+        });
+
+        queryClient.setQueryData(externalTokenQueryKey({ address, network, currency }), tokenData);
```

Comment:
do we need to worry about setting a stale time or anything here? i have no idea just curious

Reply:
i think ideally we'd want to handle it, but for right now it's probably fine. The cache gets updated on every successful fetch, cache is only used as a fallback.

Reply:
actually thinking twice about it, went ahead and implemented an invalidation strategy